src/MediaWiki/Template/TemplateExpander.php

```diff
@@ -70,7 +71,8 @@ public function expand( $template ) {
 		$options = $this->parser->getOptions();
 
 		if ( !$options instanceof ParserOptions ) {
-			$options = new ParserOptions();
+			$user = RequestContext::getMain()->getUser();
+			$options = new ParserOptions( $user );
```

Comment:
Can avoid having this $user var in the whole method scope.

`new ParserOptions( RequestContext::getMain()->getUser() )`

Then again the whole if-body should be in its own `newParserOptions` function.